components/receive-amount.tsx

```diff
@@ -138,13 +133,62 @@ export default function ReceiveAmount({
       </div>
       <div>&#8776; {convertedValue}</div>
 
-      {satsForInvoice > 0 && (
-        <GenerateInvoice
-          amountInSats={satsForInvoice}
-          userWalletId={userWalletId}
-          regenerateAction={updateSatsAmount}
-        />
-      )}
+      {satsForInvoice > 0 &&
+        (errorString ? (
+          <div className="error">{errorString}</div>
+        ) : loading ? (
+          <div className="loading">{loading && "Creating Invoice"}...</div>
+        ) : !data ? null : (
```

Comment:
These nested conditions are hard to understand.

Reply:
I agree.

Reply:
This logic was previously in if statements in the `GenerateInvoice` component which is cleaner IMO.

Please keep the `GenerateInvoice` component. It defines good boundaries.

Reply:
Done.